php-checks/src/main/java/org/sonar/php/checks/utils/AbstractDuplicateBranchImplementationCheck.java

```diff
@@ -117,11 +119,32 @@ private static boolean endsWithBreak(List<StatementTree> statements) {
   }
 
   private static boolean areAllEquivalent(List<List<StatementTree>> branches) {
+    if (branches.isEmpty()) {
+      return false;
+    }
     List<StatementTree> firstBranch = branches.get(0);
     return branches.stream().allMatch(branch -> SyntacticEquivalence.areSyntacticallyEquivalent(firstBranch, branch));
   }
 
-  protected abstract void onAllEquivalentBranches(SyntaxToken keyword);
+  private static List<List<StatementTree>> getNonTrivialBranches(List<List<StatementTree>> branches) {
+    return branches.stream()
+      .filter(AbstractDuplicateBranchImplementationCheck::isNontrivial)
+      .collect(Collectors.toList());
+  }
+
+  private static boolean isNontrivial(List<StatementTree> statements) {
+    List<StatementTree> normalizedStatements = normalize(statements);
+    if (normalizedStatements.isEmpty()) {
+      return false;
+    }
+    if (normalizedStatements.size() == 1 && normalizedStatements.get(0).is(Tree.Kind.BLOCK)) {
+      BlockTree block = (BlockTree) normalizedStatements.get(0);
+      return block.statements().size() > 1;
```

Comment:
According to [RSPEC-1871](https://jira.sonarsource.com/browse/RSPEC-1871), the exception is about lines of code, not about the number of statements.